pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -3180,12 +3180,56 @@ public TableStats getTableStats(String tableNameWithType) {
     return new TableStats(creationTime);
   }
 
+  /**
+   * Return the list of live brokers serving the corresponding table. Based on the
+   * input tableName, there can be 3 cases:
+   *
+   * 1. If the tableName has a type-suffix, then brokers for only that table-type
+   *    will be returned.
+   * 2. If the tableName doesn't have a type-suffix and there's only 1 type for that
+   *    table, then the brokers for that table-type would be returned.
+   * 3. If the tableName doesn't have a type-suffix and there are both REALTIME
+   *    and OFFLINE tables, then the intersection of the brokers for the two table-types
+   *    would be returned. Intersection is taken since the method guarantees to return
+   *    brokers which can serve the given table. In case of no type provided, it returns
+   *    broker which can serve both table-types.
+   *
+   * @param tableName name of table with or without type suffix.
+   * @return list of brokers serving the given table in the format: Broker_hostname_port.
+   * @throws TableNotFoundException when no table exists with the given name.
+   */
+  public List<String> getLiveBrokersForTable(String tableName)
+      throws TableNotFoundException {
+    TableType inputTableType = TableNameBuilder.getTableTypeFromTableName(tableName);
+    if (inputTableType != null) {
+      if (!hasTable(tableName)) {
+        throw new TableNotFoundException(String.format("Table=%s not found", tableName));
+      }
+      return getLiveBrokersForTable(tableName, inputTableType);
+    }
+    boolean hasOfflineTable = hasOfflineTable(tableName);
```

Comment:
(minor) Generate `offlineTableName` and `realtimeTableName` using `TableNameBuilder.OFFLINE.tableNameWithType(tableName)` and `TableNameBuilder.REALTIME.tableNameWithType(tableName)` which can be used to check table existence and lookup the external view to avoid generating them multiple times.